django/db/backends/utils.py

```diff
@@ -75,13 +75,26 @@ def _execute_with_wrappers(self, sql, params, many, executor):
             executor = functools.partial(wrapper, executor)
         return executor(sql, params, many, context)
 
+    def _valid_param(self, param):
+        if isinstance(param, str) and len(param.split('\x00')) > 1:
```

Comment:
Splitting params creates extra lists in memory which should not be needed, `param.count('\x00')` does not work? Actually a simple `param.find` would also do it since we just need to know if there is any null byte in there.